### DISCUSSION TITLE
Naming conflicts between r-universe and CRAN packages during dependency

### DISCUSSION THREAD

**Discussion by robchallen** [r-universe-org/help#402] [Q&A] [answered]
I've worked around this but wondering if there is a solution other than "don't do anything so foolish in the future".

I created a package in my r-universe called "tableone". There is already a CRAN package of that name (my first mistake). It would be slightly painful to rename it as people in the organisation are using it.

I wanted to depend on it from a second package (`modelfitter`). using a 

```
...
Suggests:
    tableone (>= 0.4.3)
Remotes: bristol-vaccine-centre/tableone
...
```

stanza in my DESCRIPTION file. The problem is that the CRAN `tableone` package is at version 0.13.2 so in some r-universe builds (Windows and MacOS - binary package builds maybe?) it looks like the CRAN package gets picked in preference to my `r-universe` version ([example here](https://github.com/r-universe/bristol-vaccine-centre/actions/runs/8850546236)). I guess that it can't see them as different things during dependency resolution.

I guess my options are:

1. rename my `tableone` (sigh)
2. bump my `tableone` version up past the `tableone` on CRAN `0.13.2` (flaky)
3. remove the dependency (N.B. this is what I did)

I did a bit of googling looking to see if you can specify a blacklist for a specific repository somehow  but that does not seem to be an thing.

Any other ideas?

**Comment by jeroen** [comment, marked as answer]
Yes you should rename the package, otherwise it is unusable in practice. 

R is very eager in updating packages from CRAN. Even if you get it to work locally, users will get in trouble because R will automatically "update" the package and replace it with the CRAN version on the user machine when it sees a chance to do so. 

Also other CRAN packages may depend on the CRAN version of `tableone` and import functions from it, which don't exist in your `tableone` so it turns into a big mess...

In the r-universe dashboard we try to highlight if a package conflicts with a CRAN name with a red icon and a popover, although it is not always trivial to determine if this is a different package with the same name, or just an older version:

<img width="603" alt="Screenshot 2024-05-01 at 12 09 27" src="https://github.com/r-universe-org/help/assets/216319/61d25714-c851-4111-8f8c-078a80aac385">

**Reply by robchallen** [reply]
thanks. I guessed that was going to be the answer :-)
